### ISSUE TITLE
Subscribe button label desync

### ISSUE THREAD

**Issue by SleepingFox88** [lbryio/lbry-desktop#4048]
**Bug**
The Subscribe button is suppose to show ``Following`` when not moused over and ``Unfollow`` when moused over. If you run your mouse over it and force it's label to change many times per second and then stop, it can sometimes show ``Unfollow`` when not moused over and ``Following`` when moused over.

**To Reproduce**
1. Run your mouse over it and force it's label to change many times per second and then stop
2. Check if button state is inverted
3. If no issue, return to step one

**Expected behavior**
The Subscribe button is suppose to show ``Following`` when not moused over and ``Unfollow`` when moused over.

**Screenshots**
https://streamable.com/f7t2ye

**System Configuration**

App | 0.45.0
-- | --
Daemon (lbrynet) | 0.70.0
Connected Email | seanloganking@gmail.com Update mailing preferences
Reward Eligible | No
Platform | Linux (Linux-5.4.34-1-MANJARO-x86_64-with-ManjaroLinux-20.0-Lysia)
Access Token | View

daemon details:
```
{
  "jsonrpc": "2.0",
  "result": {
    "build": "release",
    "desktop": "X-Cinnamon",
    "distro": {
      "codename": "Lysia",
      "id": "manjaro",
      "like": "arch",
      "version": "20.0",
      "version_parts": {
        "build_number": "",
        "major": "20",
        "minor": "0"
      }
    },
    "lbrynet_version": "0.70.0",
    "os_release": "5.4.34-1-MANJARO",
    "os_system": "Linux",
    "platform": "Linux-5.4.34-1-MANJARO-x86_64-with-ManjaroLinux-20.0-Lysia",
    "processor": "",
    "python_version": "3.7.7",
    "version": "0.70.0"
  }
}
```

**Comment by SleepingFox88** [comment]
I left my Installation ID out of my listed System Configuration details. Is it safe to share?

**Comment by kauffj** [comment]
@SleepingFox88 it's safe to share in the sense it couldn't be used to compromise your account or wallet, but it is information that could be used to identify your install

just keep that access token nice and safe :)

**Comment by HarshKhandeparkar** [comment]
I would like to work on this, if possible.

**Comment by HarshKhandeparkar** [comment]
I think the issue is that currently the DOM event changes the react state of the button which causes the button to re-render. But this is a bit inefficient. The react state changes can be slow and be debounced too.

What should instead happen is the DOM event callback should directly change the icon which is a bit difficult to do.

**Comment by HarshKhandeparkar** [comment]
I think that this bug should not be fixed as there is no good way to stop this from happening. We need to either force re-render the component (which is not recommended) or manipulate the DOM directly(also not recommended).

Also, too many fast DOM events will only affect performance. We should just leave it as it is for now. It is not a big issue.

**Comment by HarshKhandeparkar** [comment]
Another issue could be that both the mouse-enter and mouse-leave events trigger the same function.

**Comment by HarshKhandeparkar** [comment]
> Another issue could be that both the mouse-enter and mouse-leave events trigger the same function.

Which means that even if it gets out of sync, it cannot get re-synced.

**Comment by HarshKhandeparkar** [comment]
Okay I am working on optimizing it a little bit.

**Comment by HarshKhandeparkar** [comment]
@seanyesmunt can this one be closed?

**Comment by seanyesmunt** [comment]
Yep. Thanks for the reminder.

**Comment by HarshKhandeparkar** [comment]
I had a suggestion. I think we should add something like `Fixes #0000 (<==== replace "0000" with the issue number)` to the PR template so that the issue is closed automatically upon merging.
